v7.x: backport update to V8 5.5

targos opened the pull request:
This is a backport of the update to V8 5.5 for v7.x.

Additions are:
- Revert of the breaking UTF-8 decoder changes
- Patches to recover ABI/API compatibility with V8 5.4
- Patches to keep compatibility with all platforms supported by v7.x

/cc @nodejs/collaborators 

CI: https://ci.nodejs.org/job/node-test-commit/7510
V8 tests: https://ci.nodejs.org/job/node-test-commit-v8-linux/543/

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [ ] commit message follows commit guidelines

##### Affected core subsystem(s)

V8

MylesBorins commented:
awesome! the abi smoker is going to need an update. I'll work on it
tomorrow!

On Thu, Jan 26, 2017, 3:35 PM Michaël Zasso <notifications@github.com>
wrote:

> This is a backport of the update to V8 5.5 for v7.x.
>
> Additions are:
>
>    - Revert of the breaking UTF-8 decoder changes
>    - Patches to recover ABI/API compatibility with V8 5.4
>    - Patches to keep compatibility with all platforms supported by v7.x
>
> /cc @nodejs/collaborators
> <https://github.com/orgs/nodejs/teams/collaborators>
>
> CI: https://ci.nodejs.org/job/node-test-commit/7510
> Checklist
>
>    - make -j4 test (UNIX), or vcbuild test (Windows) passes
>    - commit message follows commit guidelines
>
> Affected core subsystem(s)
>
> V8
> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/nodejs/node/pull/11029
> Commit Summary
>
>    - deps: update V8 to 5.5.372.40
>    - deps: cherry-pick workaround for clang-3.4 ICE
>    - deps: limit regress/regress-crbug-514081 v8 test
>    - test: move test-vm-function-redefinition to parallel
>    - repl: remove workaround for function redefinition
>    - ABI: get back V8 5.4 ABI compatibility
>    - ABI: remove new fields from
>    {Indexed|Named}PropertyHandlerConfiguration
>    - ABI: Revert "[tracing] Support ConvertableToTraceFormat argument
>    type."
>    - ABI: remove tests for getOwnPropertyDescriptor and defineProperty
>    interceptors
>    - ABI: Revert breaking UTF-8 decoder changes
>    - deps: work around SmartOS 14 incompatibility
>
> File Changes
>
>    - *M* deps/v8/.gitignore
>    <https://github.com/nodejs/node/pull/11029/files#diff-0> (7)
>    - *M* deps/v8/.gn
>    <https://github.com/nodejs/node/pull/11029/files#diff-1> (5)
>    - *M* deps/v8/AUTHORS
>    <https://github.com/nodejs/node/pull/11029/files#diff-2> (4)
>    - *M* deps/v8/BUILD.gn
>    <https://github.com/nodejs/node/pull/11029/files#diff-3> (468)
>    - *M* deps/v8/ChangeLog
>    <https://github.com/nodejs/node/pull/11029/files#diff-4> (1984)
>    - *M* deps/v8/DEPS
>    <https://github.com/nodejs/node/pull/11029/files#diff-5> (84)
>    - *M* deps/v8/OWNERS
>    <https://github.com/nodejs/node/pull/11029/files#diff-6> (1)
>    - *M* deps/v8/PRESUBMIT.py
>    <https://github.com/nodejs/node/pull/11029/files#diff-7> (33)
>    - *M* deps/v8/base/trace_event/common/trace_event_common.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-8> (97)
>    - *M* deps/v8/build_overrides/build.gni
>    <https://github.com/nodejs/node/pull/11029/files#diff-9> (8)
>    - *M* deps/v8/build_overrides/v8.gni
>    <https://github.com/nodejs/node/pull/11029/files#diff-10> (16)
>    - *D* deps/v8/gypfiles/config/win/msvs_dependencies.isolate
>    <https://github.com/nodejs/node/pull/11029/files#diff-11> (77)
>    - *M* deps/v8/gypfiles/get_landmines.py
>    <https://github.com/nodejs/node/pull/11029/files#diff-12> (2)
>    - *M* deps/v8/gypfiles/gyp_v8
>    <https://github.com/nodejs/node/pull/11029/files#diff-13> (14)
>    - *M* deps/v8/gypfiles/standalone.gypi
>    <https://github.com/nodejs/node/pull/11029/files#diff-14> (6)
>    - *A* deps/v8/include/DEPS
>    <https://github.com/nodejs/node/pull/11029/files#diff-15> (4)
>    - *M* deps/v8/include/OWNERS
>    <https://github.com/nodejs/node/pull/11029/files#diff-16> (5)
>    - *M* deps/v8/include/libplatform/v8-tracing.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-17> (16)
>    - *A* deps/v8/include/v8-inspector-protocol.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-18> (13)
>    - *A* deps/v8/include/v8-inspector.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-19> (267)
>    - *M* deps/v8/include/v8-platform.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-20> (13)
>    - *M* deps/v8/include/v8-profiler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-21> (28)
>    - *M* deps/v8/include/v8-util.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-22> (17)
>    - *M* deps/v8/include/v8-version.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-23> (6)
>    - *M* deps/v8/include/v8.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-24> (933)
>    - *M* deps/v8/infra/mb/mb_config.pyl
>    <https://github.com/nodejs/node/pull/11029/files#diff-25> (254)
>    - *M* deps/v8/src/address-map.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-26> (2)
>    - *M* deps/v8/src/address-map.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-27> (4)
>    - *M* deps/v8/src/allocation.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-28> (6)
>    - *M* deps/v8/src/api-arguments-inl.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-29> (41)
>    - *M* deps/v8/src/api-arguments.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-30> (4)
>    - *M* deps/v8/src/api-arguments.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-31> (7)
>    - *M* deps/v8/src/api-natives.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-32> (92)
>    - *M* deps/v8/src/api.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-33> (603)
>    - *M* deps/v8/src/api.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-34> (8)
>    - *M* deps/v8/src/arguments.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-35> (7)
>    - *D* deps/v8/src/arm/OWNERS
>    <https://github.com/nodejs/node/pull/11029/files#diff-36> (1)
>    - *M* deps/v8/src/arm/assembler-arm-inl.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-37> (2)
>    - *M* deps/v8/src/arm/assembler-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-38> (576)
>    - *M* deps/v8/src/arm/assembler-arm.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-39> (27)
>    - *M* deps/v8/src/arm/code-stubs-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-40> (375)
>    - *M* deps/v8/src/arm/codegen-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-41> (2)
>    - *M* deps/v8/src/arm/constants-arm.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-42> (24)
>    - *M* deps/v8/src/arm/deoptimizer-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-43> (25)
>    - *M* deps/v8/src/arm/disasm-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-44> (86)
>    - *M* deps/v8/src/arm/interface-descriptors-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-45> (26)
>    - *M* deps/v8/src/arm/macro-assembler-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-46> (185)
>    - *M* deps/v8/src/arm/macro-assembler-arm.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-47> (52)
>    - *M* deps/v8/src/arm/simulator-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-48> (170)
>    - *M* deps/v8/src/arm/simulator-arm.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-49> (19)
>    - *D* deps/v8/src/arm64/OWNERS
>    <https://github.com/nodejs/node/pull/11029/files#diff-50> (1)
>    - *M* deps/v8/src/arm64/code-stubs-arm64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-51> (405)
>    - *M* deps/v8/src/arm64/interface-descriptors-arm64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-52> (26)
>    - *M* deps/v8/src/arm64/macro-assembler-arm64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-53> (9)
>    - *M* deps/v8/src/arm64/macro-assembler-arm64.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-54> (12)
>    - *M* deps/v8/src/arm64/simulator-arm64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-55> (3)
>    - *M* deps/v8/src/arm64/simulator-arm64.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-56> (5)
>    - *M* deps/v8/src/asmjs/asm-js.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-57> (54)
>    - *M* deps/v8/src/asmjs/asm-js.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-58> (19)
>    - *M* deps/v8/src/asmjs/asm-typer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-59> (69)
>    - *M* deps/v8/src/asmjs/asm-typer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-60> (26)
>    - *M* deps/v8/src/asmjs/asm-types.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-61> (4)
>    - *M* deps/v8/src/asmjs/asm-wasm-builder.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-62> (448)
>    - *M* deps/v8/src/asmjs/asm-wasm-builder.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-63> (7)
>    - *M* deps/v8/src/assembler.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-64> (49)
>    - *M* deps/v8/src/assembler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-65> (19)
>    - *M* deps/v8/src/assert-scope.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-66> (8)
>    - *M* deps/v8/src/ast/OWNERS
>    <https://github.com/nodejs/node/pull/11029/files#diff-67> (1)
>    - *M* deps/v8/src/ast/ast-expression-rewriter.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-68> (9)
>    - *M* deps/v8/src/ast/ast-expression-rewriter.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-69> (4)
>    - *M* deps/v8/src/ast/ast-literal-reindexer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-70> (9)
>    - *M* deps/v8/src/ast/ast-literal-reindexer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-71> (2)
>    - *M* deps/v8/src/ast/ast-numbering.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-72> (63)
>    - *M* deps/v8/src/ast/ast-traversal-visitor.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-73> (4)
>    - *M* deps/v8/src/ast/ast-type-bounds.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-74> (14)
>    - *R* deps/v8/src/ast/ast-types.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-75> (465)
>    - *A* deps/v8/src/ast/ast-types.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-76> (1024)
>    - *M* deps/v8/src/ast/ast-value-factory.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-77> (68)
>    - *M* deps/v8/src/ast/ast-value-factory.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-78> (27)
>    - *M* deps/v8/src/ast/ast.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-79> (142)
>    - *M* deps/v8/src/ast/ast.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-80> (653)
>    - *A* deps/v8/src/ast/compile-time-value.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-81> (56)
>    - *A* deps/v8/src/ast/compile-time-value.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-82> (45)
>    - *M* deps/v8/src/ast/context-slot-cache.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-83> (7)
>    - *M* deps/v8/src/ast/context-slot-cache.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-84> (1)
>    - *M* deps/v8/src/ast/modules.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-85> (248)
>    - *M* deps/v8/src/ast/modules.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-86> (124)
>    - *M* deps/v8/src/ast/prettyprinter.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-87> (75)
>    - *M* deps/v8/src/ast/prettyprinter.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-88> (3)
>    - *M* deps/v8/src/ast/scopeinfo.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-89> (666)
>    - *D* deps/v8/src/ast/scopeinfo.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-90> (18)
>    - *M* deps/v8/src/ast/scopes.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-91> (0)
>    - *M* deps/v8/src/ast/scopes.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-92> (0)
>    - *M* deps/v8/src/ast/variables.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-93> (0)
>    - *M* deps/v8/src/ast/variables.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-94> (0)
>    - *M* deps/v8/src/background-parsing-task.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-95> (0)
>    - *M* deps/v8/src/background-parsing-task.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-96> (0)
>    - *M* deps/v8/src/bailout-reason.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-97> (0)
>    - *M* deps/v8/src/base.isolate
>    <https://github.com/nodejs/node/pull/11029/files#diff-98> (0)
>    - *M* deps/v8/src/base/accounting-allocator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-99> (0)
>    - *M* deps/v8/src/base/accounting-allocator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-100> (0)
>    - *M* deps/v8/src/base/atomic-utils.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-101> (0)
>    - *M* deps/v8/src/base/build_config.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-102> (0)
>    - *M* deps/v8/src/base/hashmap-entry.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-103> (0)
>    - *M* deps/v8/src/base/hashmap.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-104> (0)
>    - *M* deps/v8/src/base/macros.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-105> (0)
>    - *M* deps/v8/src/base/platform/platform-macos.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-106> (0)
>    - *M* deps/v8/src/basic-block-profiler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-107> (0)
>    - *M* deps/v8/src/bit-vector.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-108> (0)
>    - *M* deps/v8/src/bootstrapper.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-109> (0)
>    - *M* deps/v8/src/builtins/arm/builtins-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-110> (0)
>    - *M* deps/v8/src/builtins/arm64/builtins-arm64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-111> (0)
>    - *M* deps/v8/src/builtins/builtins-array.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-112> (0)
>    - *M* deps/v8/src/builtins/builtins-callsite.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-113> (0)
>    - *M* deps/v8/src/builtins/builtins-conversion.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-114> (0)
>    - *M* deps/v8/src/builtins/builtins-dataview.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-115> (0)
>    - *M* deps/v8/src/builtins/builtins-date.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-116> (0)
>    - *M* deps/v8/src/builtins/builtins-global.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-117> (0)
>    - *M* deps/v8/src/builtins/builtins-handler.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-118> (0)
>    - *M* deps/v8/src/builtins/builtins-internal.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-119> (0)
>    - *M* deps/v8/src/builtins/builtins-interpreter.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-120> (0)
>    - *M* deps/v8/src/builtins/builtins-iterator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-121> (0)
>    - *M* deps/v8/src/builtins/builtins-number.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-122> (0)
>    - *M* deps/v8/src/builtins/builtins-object.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-123> (0)
>    - *M* deps/v8/src/builtins/builtins-regexp.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-124> (0)
>    - *M* deps/v8/src/builtins/builtins-sharedarraybuffer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-125> (0)
>    - *M* deps/v8/src/builtins/builtins-string.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-126> (0)
>    - *M* deps/v8/src/builtins/builtins-utils.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-127> (0)
>    - *M* deps/v8/src/builtins/builtins.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-128> (0)
>    - *M* deps/v8/src/builtins/ia32/builtins-ia32.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-129> (0)
>    - *M* deps/v8/src/builtins/mips/builtins-mips.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-130> (0)
>    - *M* deps/v8/src/builtins/mips64/builtins-mips64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-131> (0)
>    - *M* deps/v8/src/builtins/ppc/builtins-ppc.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-132> (0)
>    - *M* deps/v8/src/builtins/s390/builtins-s390.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-133> (0)
>    - *M* deps/v8/src/builtins/x64/builtins-x64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-134> (0)
>    - *M* deps/v8/src/builtins/x87/builtins-x87.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-135> (0)
>    - *M* deps/v8/src/checks.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-136> (0)
>    - *M* deps/v8/src/code-events.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-137> (0)
>    - *M* deps/v8/src/code-factory.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-138> (0)
>    - *M* deps/v8/src/code-factory.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-139> (0)
>    - *M* deps/v8/src/code-stub-assembler.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-140> (0)
>    - *M* deps/v8/src/code-stub-assembler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-141> (0)
>    - *M* deps/v8/src/code-stubs-hydrogen.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-142> (0)
>    - *M* deps/v8/src/code-stubs.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-143> (0)
>    - *M* deps/v8/src/code-stubs.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-144> (0)
>    - *M* deps/v8/src/codegen.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-145> (0)
>    - *M* deps/v8/src/collector.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-146> (0)
>    - *M* deps/v8/src/compilation-dependencies.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-147> (0)
>    - *M* deps/v8/src/compilation-info.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-148> (0)
>    - *M* deps/v8/src/compilation-info.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-149> (0)
>    - *M* deps/v8/src/compiler-dispatcher/compiler-dispatcher-job.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-150> (0)
>    - *M* deps/v8/src/compiler-dispatcher/compiler-dispatcher-job.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-151> (0)
>    - *M* deps/v8/src/compiler-dispatcher/optimizing-compile-dispatcher.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-152> (0)
>    - *M* deps/v8/src/compiler.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-153> (0)
>    - *M* deps/v8/src/compiler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-154> (0)
>    - *M* deps/v8/src/compiler/access-builder.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-155> (0)
>    - *M* deps/v8/src/compiler/access-builder.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-156> (0)
>    - *M* deps/v8/src/compiler/access-info.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-157> (0)
>    - *M* deps/v8/src/compiler/access-info.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-158> (0)
>    - *M* deps/v8/src/compiler/all-nodes.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-159> (0)
>    - *M* deps/v8/src/compiler/all-nodes.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-160> (0)
>    - *M* deps/v8/src/compiler/arm/code-generator-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-161> (0)
>    - *M* deps/v8/src/compiler/arm/instruction-selector-arm.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-162> (0)
>    - *M* deps/v8/src/compiler/arm64/code-generator-arm64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-163> (0)
>    - *M* deps/v8/src/compiler/arm64/instruction-selector-arm64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-164> (0)
>    - *M* deps/v8/src/compiler/ast-graph-builder.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-165> (0)
>    - *M* deps/v8/src/compiler/ast-graph-builder.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-166> (0)
>    - *M* deps/v8/src/compiler/ast-loop-assignment-analyzer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-167> (0)
>    - *M* deps/v8/src/compiler/ast-loop-assignment-analyzer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-168> (0)
>    - *M* deps/v8/src/compiler/basic-block-instrumentor.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-169> (0)
>    - *M* deps/v8/src/compiler/bytecode-graph-builder.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-170> (0)
>    - *M* deps/v8/src/compiler/bytecode-graph-builder.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-171> (0)
>    - *M* deps/v8/src/compiler/bytecode-loop-analysis.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-172> (0)
>    - *M* deps/v8/src/compiler/c-linkage.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-173> (0)
>    - *M* deps/v8/src/compiler/code-assembler.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-174> (0)
>    - *M* deps/v8/src/compiler/code-assembler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-175> (0)
>    - *M* deps/v8/src/compiler/code-generator-impl.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-176> (0)
>    - *M* deps/v8/src/compiler/code-generator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-177> (0)
>    - *M* deps/v8/src/compiler/code-generator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-178> (0)
>    - *M* deps/v8/src/compiler/common-operator-reducer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-179> (0)
>    - *M* deps/v8/src/compiler/common-operator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-180> (0)
>    - *M* deps/v8/src/compiler/common-operator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-181> (0)
>    - *M* deps/v8/src/compiler/control-equivalence.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-182> (0)
>    - *M* deps/v8/src/compiler/control-flow-optimizer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-183> (0)
>    - *M* deps/v8/src/compiler/effect-control-linearizer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-184> (0)
>    - *M* deps/v8/src/compiler/effect-control-linearizer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-185> (0)
>    - *M* deps/v8/src/compiler/escape-analysis-reducer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-186> (0)
>    - *M* deps/v8/src/compiler/escape-analysis.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-187> (0)
>    - *M* deps/v8/src/compiler/escape-analysis.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-188> (0)
>    - *M* deps/v8/src/compiler/graph-reducer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-189> (0)
>    - *M* deps/v8/src/compiler/graph-visualizer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-190> (0)
>    - *M* deps/v8/src/compiler/graph.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-191> (0)
>    - *M* deps/v8/src/compiler/ia32/code-generator-ia32.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-192> (0)
>    - *M* deps/v8/src/compiler/ia32/instruction-scheduler-ia32.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-193> (0)
>    - *M* deps/v8/src/compiler/ia32/instruction-selector-ia32.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-194> (0)
>    - *M* deps/v8/src/compiler/instruction-codes.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-195> (0)
>    - *M* deps/v8/src/compiler/instruction-scheduler.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-196> (0)
>    - *M* deps/v8/src/compiler/instruction-scheduler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-197> (0)
>    - *M* deps/v8/src/compiler/instruction-selector-impl.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-198> (0)
>    - *M* deps/v8/src/compiler/instruction-selector.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-199> (0)
>    - *M* deps/v8/src/compiler/instruction-selector.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-200> (0)
>    - *M* deps/v8/src/compiler/instruction.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-201> (0)
>    - *M* deps/v8/src/compiler/instruction.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-202> (0)
>    - *M* deps/v8/src/compiler/int64-lowering.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-203> (0)
>    - *M* deps/v8/src/compiler/int64-lowering.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-204> (0)
>    - *M* deps/v8/src/compiler/js-builtin-reducer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-205> (0)
>    - *M* deps/v8/src/compiler/js-builtin-reducer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-206> (0)
>    - *M* deps/v8/src/compiler/js-call-reducer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-207> (0)
>    - *M* deps/v8/src/compiler/js-call-reducer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-208> (0)
>    - *M* deps/v8/src/compiler/js-create-lowering.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-209> (0)
>    - *M* deps/v8/src/compiler/js-create-lowering.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-210> (0)
>    - *M* deps/v8/src/compiler/js-generic-lowering.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-211> (0)
>    - *M* deps/v8/src/compiler/js-global-object-specialization.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-212> (0)
>    - *M* deps/v8/src/compiler/js-global-object-specialization.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-213> (0)
>    - *M* deps/v8/src/compiler/js-inlining-heuristic.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-214> (0)
>    - *M* deps/v8/src/compiler/js-inlining-heuristic.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-215> (0)
>    - *M* deps/v8/src/compiler/js-inlining.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-216> (0)
>    - *M* deps/v8/src/compiler/js-inlining.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-217> (0)
>    - *M* deps/v8/src/compiler/js-intrinsic-lowering.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-218> (0)
>    - *M* deps/v8/src/compiler/js-native-context-specialization.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-219> (0)
>    - *M* deps/v8/src/compiler/js-native-context-specialization.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-220> (0)
>    - *M* deps/v8/src/compiler/js-operator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-221> (0)
>    - *M* deps/v8/src/compiler/js-operator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-222> (0)
>    - *M* deps/v8/src/compiler/js-typed-lowering.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-223> (0)
>    - *M* deps/v8/src/compiler/js-typed-lowering.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-224> (0)
>    - *M* deps/v8/src/compiler/linkage.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-225> (0)
>    - *M* deps/v8/src/compiler/linkage.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-226> (0)
>    - *M* deps/v8/src/compiler/live-range-separator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-227> (0)
>    - *M* deps/v8/src/compiler/live-range-separator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-228> (0)
>    - *M* deps/v8/src/compiler/liveness-analyzer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-229> (0)
>    - *M* deps/v8/src/compiler/load-elimination.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-230> (0)
>    - *M* deps/v8/src/compiler/load-elimination.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-231> (0)
>    - *M* deps/v8/src/compiler/loop-analysis.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-232> (0)
>    - *M* deps/v8/src/compiler/loop-analysis.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-233> (0)
>    - *M* deps/v8/src/compiler/loop-peeling.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-234> (0)
>    - *M* deps/v8/src/compiler/loop-variable-optimizer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-235> (0)
>    - *M* deps/v8/src/compiler/loop-variable-optimizer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-236> (0)
>    - *M* deps/v8/src/compiler/machine-graph-verifier.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-237> (0)
>    - *M* deps/v8/src/compiler/machine-graph-verifier.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-238> (0)
>    - *M* deps/v8/src/compiler/machine-operator-reducer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-239> (0)
>    - *M* deps/v8/src/compiler/machine-operator-reducer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-240> (0)
>    - *M* deps/v8/src/compiler/machine-operator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-241> (0)
>    - *M* deps/v8/src/compiler/machine-operator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-242> (0)
>    - *M* deps/v8/src/compiler/memory-optimizer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-243> (0)
>    - *M* deps/v8/src/compiler/memory-optimizer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-244> (0)
>    - *M* deps/v8/src/compiler/mips/code-generator-mips.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-245> (0)
>    - *M* deps/v8/src/compiler/mips/instruction-codes-mips.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-246> (0)
>    - *M* deps/v8/src/compiler/mips/instruction-selector-mips.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-247> (0)
>    - *M* deps/v8/src/compiler/mips64/code-generator-mips64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-248> (0)
>    - *M* deps/v8/src/compiler/mips64/instruction-codes-mips64.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-249> (0)
>    - *M* deps/v8/src/compiler/mips64/instruction-selector-mips64.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-250> (0)
>    - *M* deps/v8/src/compiler/move-optimizer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-251> (0)
>    - *M* deps/v8/src/compiler/move-optimizer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-252> (0)
>    - *M* deps/v8/src/compiler/node-aux-data.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-253> (0)
>    - *M* deps/v8/src/compiler/node-cache.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-254> (0)
>    - *M* deps/v8/src/compiler/node-matchers.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-255> (0)
>    - *M* deps/v8/src/compiler/node-properties.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-256> (0)
>    - *M* deps/v8/src/compiler/node.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-257> (0)
>    - *M* deps/v8/src/compiler/opcodes.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-258> (0)
>    - *M* deps/v8/src/compiler/operation-typer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-259> (0)
>    - *M* deps/v8/src/compiler/operation-typer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-260> (0)
>    - *M* deps/v8/src/compiler/operator-properties.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-261> (0)
>    - *M* deps/v8/src/compiler/operator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-262> (0)
>    - *M* deps/v8/src/compiler/osr.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-263> (0)
>    - *M* deps/v8/src/compiler/osr.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-264> (0)
>    - *M* deps/v8/src/compiler/pipeline-statistics.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-265> (0)
>    - *M* deps/v8/src/compiler/pipeline.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-266> (0)
>    - *M* deps/v8/src/compiler/ppc/code-generator-ppc.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-267> (0)
>    - *M* deps/v8/src/compiler/ppc/instruction-selector-ppc.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-268> (0)
>    - *M* deps/v8/src/compiler/raw-machine-assembler.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-269> (0)
>    - *M* deps/v8/src/compiler/raw-machine-assembler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-270> (0)
>    - *M* deps/v8/src/compiler/redundancy-elimination.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-271> (0)
>    - *M* deps/v8/src/compiler/register-allocator-verifier.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-272> (0)
>    - *M* deps/v8/src/compiler/register-allocator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-273> (0)
>    - *M* deps/v8/src/compiler/register-allocator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-274> (0)
>    - *M* deps/v8/src/compiler/representation-change.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-275> (0)
>    - *M* deps/v8/src/compiler/representation-change.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-276> (0)
>    - *M* deps/v8/src/compiler/s390/code-generator-s390.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-277> (0)
>    - *M* deps/v8/src/compiler/s390/instruction-selector-s390.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-278> (0)
>    - *M* deps/v8/src/compiler/schedule.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-279> (0)
>    - *M* deps/v8/src/compiler/scheduler.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-280> (0)
>    - *M* deps/v8/src/compiler/scheduler.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-281> (0)
>    - *M* deps/v8/src/compiler/simplified-lowering.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-282> (0)
>    - *M* deps/v8/src/compiler/simplified-lowering.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-283> (0)
>    - *M* deps/v8/src/compiler/simplified-operator-reducer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-284> (0)
>    - *M* deps/v8/src/compiler/simplified-operator.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-285> (0)
>    - *M* deps/v8/src/compiler/simplified-operator.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-286> (0)
>    - *M* deps/v8/src/compiler/state-values-utils.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-287> (0)
>    - *M* deps/v8/src/compiler/store-store-elimination.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-288> (0)
>    - *M* deps/v8/src/compiler/store-store-elimination.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-289> (0)
>    - *R* deps/v8/src/compiler/type-cache.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-290> (0)
>    - *M* deps/v8/src/compiler/type-cache.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-291> (0)
>    - *M* deps/v8/src/compiler/type-hint-analyzer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-292> (0)
>    - *M* deps/v8/src/compiler/type-hint-analyzer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-293> (0)
>    - *M* deps/v8/src/compiler/typed-optimization.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-294> (0)
>    - *M* deps/v8/src/compiler/typed-optimization.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-295> (0)
>    - *M* deps/v8/src/compiler/typer.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-296> (0)
>    - *M* deps/v8/src/compiler/typer.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-297> (0)
>    - *M* deps/v8/src/compiler/types.cc
>    <https://github.com/nodejs/node/pull/11029/files#diff-298> (0)
>    - *R* deps/v8/src/compiler/types.h
>    <https://github.com/nodejs/node/pull/11029/files#diff-299> (0)
>
> Patch Links:
>
>    - https://github.com/nodejs/node/pull/11029.patch
>    - https://github.com/nodejs/node/pull/11029.diff
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/nodejs/node/pull/11029>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAecVzaxiv8lL42ovsAAJXzFbPjctpeEks5rWS2ogaJpZM4LvRsi>
> .
>

ofrobots commented:
This is heroic; kudos for putting it together!

misterdjules commented:
Thank you @targos!

mcollina commented:
Does this include https://github.com/nodejs/node/pull/9730 as well? It might need to be reapplied.

targos commented:
@mcollina Yes it is included. Ref:https://github.com/v8/v8/commit/a3197fa79f36a9eaddb73db3d9d21b41c210485c

mcollina commented:
Thanks!

targos commented:
@MylesBorins is https://ci.nodejs.org/view/Node.js-citgm/job/citgm-abi-smoker/ the job to launch?

Trott commented:
Needs a rebase. Also, do any of the Collaborators giving this emoticon upvotes and positive comments feel comfortable enough approving it? 1500+ files changed. Personally, I wouldn't know where to begin to even spot-check this sort of thing before dropping in a rubber-stamp approval. Paging @nodejs/v8 for help.

rom1504 commented:
@Trott if you look at all commits of this PR except "deps: update V8 to 5.5.372.40" then actually little changes were made ;)

bricss commented:
I believe that V8 dev team already approved it, also there is a pile of tests for this. Just merge it. 😃

MylesBorins commented:
I'm going to spend some time tomorrow getting the abi smoker up and running again. Thanks for the patience everyone :D

MylesBorins commented:
Rebased the rebuild job and fingers crossed it should be working

--> https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-rebuild/3/

Will dig into results tomorrow

addaleax commented on deps/v8/include/v8.h:
```diff
@@ -5574,6 +6169,11 @@ class V8_EXPORT EmbedderHeapTracer {
+  /**
+   * Returns the number of wrappers that are still to be traced by the embedder.
+   */
+  virtual size_t NumberOfWrappersToTrace() { return 0; }
```
Just for my own understanding – it doesn’t make sense for addons to set an `EmbedderHeapTracer`, which is why we don’t care about these breakages, right?

bnoordhuis replied:
I suppose it is technically possible for an add-on to call `Isolate::SetEmbedderHeapTracer()` but I can't see that being useful: an add-on won't know how to trace objects it didn't create itself.

(Neither does core know about objects created by add-ons.  This API doesn't seem all that usable inside the node ecosystem, it seems to assume the embedder is a single monolithic entity.)

addaleax replied:
@bnoordhuis Thanks for confirming!

addaleax commented on deps/v8/include/v8.h:
```diff
@@ -6857,8 +7460,6 @@ class V8_EXPORT V8 {
   static Local<Value> GetEternal(Isolate* isolate, int index);
 
-  static void RegisterExternallyReferencedObject(internal::Object** object,
-                                                 internal::Isolate* isolate);
```
(ditto)

targos commented:
Rebased and squashed commits about ABI compatibility.
CI: https://ci.nodejs.org/job/node-test-pull-request/6239/

@MylesBorins What's the conclusion with the CITGM ABI smoker?

MylesBorins commented:
One more run of the ABI smoker. So far everything looked good except for a couple of odd failures on Ubuntu 1404. I have a beefy VM spun up and am going to test those ones manually

targos commented:
Thank you @MylesBorins!
Only CI failure is the flaky sequential/test-child-process-pass-fd on [fedora 24](https://github.com/nodejs/node/issues/11041). Going to land soon.
